.github/workflows/rust.yml

```diff
@@ -0,0 +1,77 @@
+name: CI - Rust
+
+on:
+  workflow_call:
+
+jobs:
+  check:
+    name: Check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          git config --global url."https://${{ secrets.CI_ACCESS_TOKEN }}@github.com/".insteadOf ssh://git@github.com
+        shell: bash
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: check
+
+  test:
+    name: Test Suite
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: |
+          git config --global url."https://${{ secrets.CI_ACCESS_TOKEN }}@github.com/".insteadOf ssh://git@github.com
+        shell: bash
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: test
+
+  fmt:
+    name: Rustfmt
```

Comment:
I am ok with a pipeline validating that the code is well formatted but it shouldn't change the format - does this just validate?

Reply:
oh -- good point, will fix

Reply:
(actually, yes it should just run it in `check` mode exiting with non-zero status if not correctly formatted)